### ISSUE TITLE
Implementation of pubsub seems incorrect

### ISSUE THREAD

**Issue by jamesporritt** [locationlabs/mockredis#115]
The pubsub attribute in mockredis/client.py seems to be defined as a defaultdict that stores the lines passed by publish, whereas it should be a method that creates an object of PubSub(), that handles the subscription to the channel. We would still need that defaultdict under the cover and linked to the mocked PubSub instances that are handed out.

**Comment by alertedsnake** [comment]
I agree, this current implementation is completely broken, so I can't use this module at all.

Is the pubsub feature on the roadmap to be completed soon?

**Comment by MPalarya** [comment]
@alertedsnake 5 years later - thank you!
